cpp/ycm/IdentifierUtils.cpp

```diff
@@ -114,38 +113,34 @@ const char *const NOT_FOUND = "YCMFOOBAR_NOT_FOUND";
 FiletypeIdentifierMap ExtractIdentifiersFromTagsFile(
   const fs::path &path_to_tag_file ) {
   FiletypeIdentifierMap filetype_identifier_map;
-  std::string tags_file_contents;
+  std::vector< std::string > tags_file_contents;
 
   try {
     tags_file_contents = ReadUtf8File( path_to_tag_file );
   } catch ( ... ) {
     return filetype_identifier_map;
   }
 
-  std::string::const_iterator start = tags_file_contents.begin();
-  std::string::const_iterator end   = tags_file_contents.end();
-
-  boost::smatch matches;
-  const boost::regex expression( TAG_REGEX );
-  const boost::match_flag_type options = boost::match_not_dot_newline;
-
-  while ( boost::regex_search( start, end, matches, expression, options ) ) {
-    start = matches[ 0 ].second;
-
-    std::string language( matches[ 3 ] );
-    std::string filetype = FindWithDefault( LANG_TO_FILETYPE,
-                                            language.c_str(),
-                                            NOT_FOUND );
-
-    if ( filetype == NOT_FOUND )
-      continue;
-
-    std::string identifier( matches[ 1 ] );
-    fs::path path( matches[ 2 ].str() );
-    path = fs::absolute( path, path_to_tag_file.parent_path() )
-           .make_preferred();
-
-    filetype_identifier_map[ filetype ][ path.string() ].push_back( identifier );
+  std::smatch matches;
+  const std::regex expression( TAG_REGEX );
+
+  for ( auto line : tags_file_contents ) {
+    if ( std::regex_search( line, matches, expression ) ) {
+      std::string language( matches[ 3 ] );
+      std::string filetype = FindWithDefault( LANG_TO_FILETYPE,
+                                              language.c_str(),
+                                              NOT_FOUND );
+      if ( filetype == NOT_FOUND )
+        continue;
+
+      std::string identifier( matches[ 1 ] );
+      fs::path path( matches[ 2 ].str() );
+      path = fs::absolute( path, path_to_tag_file.parent_path() )
+              .make_preferred();
+
+      filetype_identifier_map[ filetype ][ path.string() ]
+              .push_back( identifier );
```

Comment:
You can use `.emplace_back(std::move(identifier));` as it is the last usage of `identifier`.